primitives/src/v2/mod.rs

```diff
@@ -135,6 +141,20 @@ impl From<u32> for ValidatorIndex {
 	}
 }
 
+impl From<ValidatorIndex> for usize {
+	fn from(n: ValidatorIndex) -> Self {
+		n.0 as usize
+	}
+}
+
+impl From<usize> for ValidatorIndex {
```

Comment:
Why do we need an impl for `From<usize>` if it's not guaranteed to be ok? Wouldn't a `From<u32>` not make more sense and leave it to the user to decide what to do on errors instead of generating a garbage `ValidatorIndex`? `TryFrom<usize>` and `From<u32>` seem more reasonable to me.

Reply:
Ok.

Reply:
While working on this, the error:
```
the trait bound `K: From<usize>` is not satisfied
    --> primitives/src/v2/mod.rs:1616:17
     |
1616 | #[derive(Clone, RuntimeDebug)]
     |                 ^^^^^^^^^^^^ the trait `From<usize>` is not implemented for `K`
```
came up. This is why I likely implemented the `From<usize>` trait.